### PR TITLE
docs: validate imports added to code snippets

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close: https://github.com/FuelLabs/fuels-ts/issues/1666

This PR adds a new way to generate imports within code snippets by using the `#import` flag:

```ts
// #import { AssetId };

const assetId: AssetId = {
  value: Bits256,
};
```

This final result will be:
```ts
import { AssetId } from 'fuels';

const assetId: AssetId = {
  value: Bits256,
};
```

### CODE REVIEW

**Comment by danielbate** [comment]
🚀

File: apps/docs-snippets/src/guide/cookbook/transferring-assets.test.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ describe(__filename, () => {
 
   it('should successfully transfer asset to another account', async () => {
     // #region transferring-assets-1
-    // #context import { Wallet, BN, BaseAssetId } from 'fuels';
+    // #addImport: Wallet, BN, BaseAssetId
```
Comment (arboleya) [inline apps/docs-snippets/src/guide/cookbook/transferring-assets.test.ts:29]: Is there any particular reason why you went with `#addImport a, b, c`?

I think a more standardized syntax would be more intuitive:

```ts
// #import { Wallet, BN, BaseAssetId };
```

Comment (Torres-ssf) [reply apps/docs-snippets/src/guide/cookbook/transferring-assets.test.ts:29]: @arboleya No particular reason.

But yeah, the suggested approach seems more intuitive to me as well.

File: packages/fuel-gauge/src/doc-examples.test.ts [hunk 2]
```diff
@@ -344,7 +342,7 @@ describe('Doc Examples', () => {
 
   it('can create a predicate', async () => {
     // #region predicate-basic
```
Comment (petertonysmith94) [inline packages/fuel-gauge/src/doc-examples.test.ts:344]: The `predicate-basic` context isn't used in the docs.
- Suggest to add reference in the docs to `predicate-basic`

File: packages/script/src/script.test.ts [hunk 3]
```diff
@@ -68,7 +68,7 @@ const callScript = async <TData, TResult>(
 };
 
 // #region script-init
-// #context import { Script, AbiCoder, getBytesCopy } from 'fuels';
+// #import { ScriptRequest, getBytesCopy }
```
Comment (petertonysmith94) [inline packages/script/src/script.test.ts:71]: Missing `Interface` import

File: packages/wallet/src/wallet-unlocked.test.ts [hunk 4]
```diff
@@ -50,7 +50,7 @@ describe('WalletUnlocked', () => {
 
   it('Sign a transaction using wallet instance', async () => {
     // #region wallet-transaction-signing
-    // #context import { WalletUnlocked, hashMessage, Signer} from 'fuels';
+    // #import { WalletUnlocked, Signer }
```
Comment (petertonysmith94) [inline packages/wallet/src/wallet-unlocked.test.ts:53]: Missing `Provider` import

**Review by petertonysmith94 — COMMENTED** [review]
Just had a quick look through 👍🏼 
A couple a small issues 😄

File: packages/script/src/script.test.ts (hunk 3 above)
Comment (Torres-ssf) [reply packages/script/src/script.test.ts:71]: This will be addressed on another PR.
> ref: https://github.com/FuelLabs/fuels-ts/issues/1550

**Review by danielbate — COMMENTED** [review]
Nice work, especially enjoyed the test cases in `extractImports.test.ts`. 🧙🏻 

I'll admit it wasn't easy reading through `extractImports.ts`, I don't necessarily think it is something to do with how you've written it, but this kind of file parsing and line matching is never pleasant. Potentially a nit on my side, but I think it could benefit by making it more verbose, some docs on the regexes and docs for some of the real specific code. As an example:
```ts
if (ignoreMatches && ignoreMatches[1]) {
        const importsToIgnore = ignoreMatches[1].split(',').map((item) => item.trim());
...
```
What is the condition we are checking for here and what is being split, to return in what format?

But the test cases made it a very easy to understand, maybe more functions to make it more independently testable in places? Your work in #1668 is a good example of this.

File: apps/docs-snippets/src/guide/contracts/contract-balance.test.ts [hunk 5]
```diff
@@ -15,8 +15,7 @@ describe(__filename, () => {
 
   it('should successfully get a contract balance', async () => {
     // #region contract-balance-3
-    // #context import type { AssetId } from 'fuels';
-    // #context import { Wallet, BN, BaseAssetId } from 'fuels';
+    // #import { AssetId, Wallet, BN, BaseAssetId }
```
Comment (arboleya) [inline apps/docs-snippets/src/guide/contracts/contract-balance.test.ts:18]: Just reinforcing what I said already, unless there's a reason behind it, I suggest keeping these imports with an ending semi-colon to keep the standards and avoid variations and unnecessary questions.

```
// #import { AssetId, Wallet, BN, BaseAssetId };
```

File: apps/docs/.vitepress/plugins/utils/extractImports.test.ts [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+
+describe('extractImports', () => {
+  describe('parseIgnoreImportFlags', () => {
+    it('should return an empty set for no ignore flags', () => {
+      const lines = ['import { something } from "somewhere";', 'const x = 5;'];
+      expect(parseIgnoreImportFlags(lines)).toEqual(new Set());
+    });
+
+    it('should correctly parse a single ignore flag', () => {
+      const lines = ['// #ignore { testImport }'];
```
Comment (arboleya) [inline apps/docs/.vitepress/plugins/utils/extractImports.test.ts:23]: I couldn't find any snipped using this `#ignore` variation.

I suggest adding clear docs around `#import` and `#ignore` on the PR description.

**Review by arboleya — CHANGES_REQUESTED** [review]

File: apps/docs/.vitepress/plugins/utils/extractImports.test.ts (hunk 6 above)
Comment (Torres-ssf) [reply apps/docs/.vitepress/plugins/utils/extractImports.test.ts]: @arboleya I've removed the `#ignore` import functionality since it was not being used.

**Review by arboleya — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]
Looks good!

- In the PR description, can you also mention the fact that this PR adds _validation_ that checks for the validity of the imports in our docs snippets?
- The FE conditional review workflow seems to be broken on this PR

**Pull request merged**